sklearn/discriminant_analysis.py

```diff
@@ -358,7 +358,9 @@ def _solve_lsqr(self, X, y, shrinkage, covariance_estimator):
         self.means_ = _class_means(X, y)
         self.covariance_ = _class_cov(X, y, self.priors_, shrinkage,
                                       covariance_estimator)
-        self.coef_ = linalg.lstsq(self.covariance_, self.means_.T)[0].T
+        self.coef_ = linalg.lstsq(self.covariance_,
```

Comment:
I think covariance_ could be non finite

Reply:
I have some issue thinking when the class covariance could become infinite. @mathcass do you see a use case, to see if we would catch it before to call the solver.